syntax = "proto3";

message SWTPoint_buf {
    int32 x = 1;
    int32 y = 2;
    float SWT = 3;
}

message Component {
    repeated SWTPoint_buf points = 1;
    int32 maxX = 2;
    int32 minX = 3;
    int32 maxY = 4;
    int32 minY = 5;
    float width = 6;
    float height = 7;
    float mean = 8;

    // features for filtering
    float SD = 9; // standard deviation
    float WV = 10; // width variation range [0, 1]
    float AR = 11; // aspect ratio range [0.1, 1]
    float OR = 12; // occupation ratio range [0.1, 1]
    float major_axis = 13;
    float minor_axis = 14;
    float characteristic_scale = 15;
    float orientation = 16;
    float center_x = 17;
    float center_y = 18;
    int32 image = 19;
    int32 id = 20;
    bool isText = 21; // 1 - is text symbol; 0 - is not the text symbol
    bool isDarkOnLight = 22;
    bool pred = 23;
}

message Components {

    repeated Component components = 1;
}